lib/workflows/email_shutoff.py

```diff
@@ -0,0 +1,124 @@
+from typing import List, Optional, Dict, Tuple
+from openstack_query.api.query_objects import ServerQuery
+from enums.query.props.server_properties import ServerProperties
+from enums.query.props.user_properties import UserProperties
+from structs.email.email_params import EmailParams
+from enums.query.query_presets import QueryPresetsGeneric, QueryPresetsDateTime
+from email_api.emailer import Emailer
+from structs.email.smtp_account import SMTPAccount
+
+
+def query_shutoff_vms(
+    cloud_account: str,
+    project_id: Optional[List[str]] = None,
+    days_threshold: int = None,
+):
+    """
+    Sends an email to a user about a shutoff (stopped) VM
+    :param cloud_account: Cloud account to use for running query
+    :param project_id: optional field to provide a list of project IDs for specific projects,
+    otherwise query all projects
+    :param days_threshold: param to query servers updated to be in shutoff state the last 30 days by default
+    :returns: Dictionary with a list of VM names grouped by user ID
+    """
+
+    query_obj = ServerQuery()
+
+    query_obj.select(ServerProperties.SERVER_NAME).where(
+        QueryPresetsGeneric.EQUAL_TO, ServerProperties.SERVER_STATUS, value="SHUTOFF"
+    ).where(
+        QueryPresetsDateTime.OLDER_THAN,
+        ServerProperties.SERVER_LAST_UPDATED_DATE,
+        days=days_threshold,
+    )
+
+    # if project ID provided as parameter
+    if project_id:
+        query_obj.run(cloud_account, as_admin=True, from_projects=project_id)
+
+    else:
+        query_obj.run(cloud_account, as_admin=True, all_projects=True)
+
+    user_query = query_obj.then("USER_QUERY", keep_previous_results=True)
+
```

Comment:
I found that this introduces a bug I had to fix in #184
This will ignore ERROR and SHUTOFF VMs which have a user_id which can't be found by running a UserQuery - either the users don't exist anymore or there's some issue in openstack. 

In these cases we should not ignore these values, but instead send these to cloud-support email

Reply:
have a look at https://github.com/stfc/st2-cloud-pack/blob/main/lib/workflows/email/send_decom_flavor_email.py#L159-L174

I split out finding user info from the original query. 
1. Group the server query by user id
2. for each user id run a separate UserQuery
3. if the UserQuery returns nothing - set "default" values for user info (set cloud-support@stfc.ac.uk) as user_email
4. if UserQuery returns something - use that